### DISCUSSION TITLE
Multiple Kserve Version in Same Kubernetes Cluster

### DISCUSSION THREAD

**Discussion by Ivikassingh** [kubeflow/kfserving#3108] [General]
Hello,

I'm aiming to maintain two versions of Kserve within a single Kubernetes cluster. The primary goal is to allow different teams or projects to utilize their preferred version of Kserve without interference.

I've attempted multiple configurations and setups to achieve this, but so far, I've encountered challenges in running both versions concurrently.

Has anyone successfully implemented such a setup? If so, can you provide guidance or best practices to achieve this?

Any help or insights would be greatly appreciated.

Thank you!

**Comment by yuzisun** [comment]
We do not support installing multiple kserve versions on a single Kubernetes cluster. It is probably not a good idea in general as different kserve versions may depend on different versions of Istio/Knative/Kubernetes. It would be really challenging to run all these combinations and haven't heard of any other Kubernetes component supporting this. 

On the other hand it is possible to run different versions of serving runtimes for a given KServe version, e.g you can still run different versions of scikit-learn server on the same kserve version by pinning the tag on inference service.
